### PR TITLE
Mark internal Lite APIs as protected

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

This PR marks some of the utilities in Lite as protected. These are ones that we consider internal and that we may want to change in the future. To avoid breaking changes in the future, we do this now. The messages in the existing deprecations get adjusted.

Proposed by @carmocca 

### Does your PR introduce any breaking changes? If yes, please list them.

No. We will consider adding these to the graveyard once the deprecation ends.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @justusschock @awaelchli @rohitgr7 @carmocca

### CODE REVIEW

File: docs/source-pytorch/accelerators/gpu_intermediate.rst [hunk 1]
```diff
@@ -491,7 +491,7 @@ Below are the possible configurations we support.
 
 Implement Your Own Distributed (DDP) training
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-If you need your own way to init PyTorch DDP you can override :meth:`pytorch_lightning.strategies.ddp.DDPStrategy.init_dist_connection`.
+If you need your own way to init PyTorch DDP you can override :meth:`pytorch_lightning.strategies.ddp.DDPStrategy._init_dist_connection`.
```
Comment (carmocca) [inline docs/source-pytorch/accelerators/gpu_intermediate.rst:494]: This method does not exist. We should probably remove this or update it

File: docs/source-pytorch/accelerators/tpu_faq.rst [hunk 2]
```diff
@@ -16,7 +16,7 @@ XLA configuration is missing?
     RuntimeError: tensorflow/compiler/xla/xla_client/computation_client.cc:273 : Missing XLA configuration
     Traceback (most recent call last):
     ...
-    File "/home/kaushikbokka/pytorch-lightning/pytorch_lightning/utilities/device_parser.py", line 125, in parse_tpu_cores
+    File "/home/kaushikbokka/pytorch-lightning/pytorch_lightning/utilities/device_parser.py", line 125, in _parse_tpu_cores
```
Comment (carmocca) [inline docs/source-pytorch/accelerators/tpu_faq.rst:19]: Accidental change

**Review by carmocca — APPROVED** [review]
Thank you! :purple_heart:

File: docs/source-pytorch/accelerators/gpu_intermediate.rst (hunk 1 above)
Comment (awaelchli) [reply docs/source-pytorch/accelerators/gpu_intermediate.rst]: updated

File: docs/source-pytorch/accelerators/tpu_faq.rst (hunk 2 above)
Comment (awaelchli) [reply docs/source-pytorch/accelerators/tpu_faq.rst]: reverted

**Review by justusschock — APPROVED** [review]

**Review by otaj — APPROVED** [review]

**Pull request merged**